app-misc/pip3line/pip3line-3.5.4.ebuild

```diff
@@ -0,0 +1,85 @@
+# Copyright 1999-2018 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python{2_7,3_{4,5,6,7}} )
+
+inherit cmake-utils python-r1
+
+DESCRIPTION="Raw bytes manipulation, transformations (decoding and more) and interception"
+HOMEPAGE="https://github.com/metrodango/pip3line"
+
+if [[ ${PV} == 9999* ]] ; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/metrodango/pip3line.git"
+	EGIT_BRANCH="master"
+else
+	SRC_URI="https://github.com/metrodango/pip3line/archive/v${PV}.tar.gz  -> ${P}.tar.gz"
+	KEYWORDS="~amd64 ~x86"
+fi
+
+LICENSE="GPL-3"
+SLOT="0"
+
+# A few comments
+# the ssl flag is just there to enable the plugin for low level crypto algorithms. 
+# It has nothing to do with the SSL/TLS protocol itself.
+
+IUSE="distorm python qscintilla ssl"
+
+RDEPEND="
+	${PYTHON_DEPS}
+	dev-qt/qtconcurrent:5
+	dev-qt/qtcore:5
+	dev-qt/qtgui:5
+	dev-qt/qtnetwork:5
+	dev-qt/qtsvg:5
+	dev-qt/qtwidgets:5
+	dev-qt/qtxmlpatterns:5
+	qscintilla? ( x11-libs/qscintilla )
+	ssl? ( dev-libs/openssl:0= )"
+
+DEPEND="${RDEPEND}
+	distorm? ( dev-vcs/git )"
+
+src_configure() {
+
+	local mycmakeargs=(
+		-DBASIC=yes
+		-DWITH_DISTORM=$(usex distorm)
+		-DWITH_OPENSSL=$(usex ssl)
+		-DWITH_SCINTILLA=$(usex qscintilla)
+	)
+
+	# distorm is statically linked, due to insufficiencies
+	# in the current distorm64 package 
+
+	if use distorm; then
+		mycmakeargs+=(-DWITH_DISTORM_LINK_STATICALLY=ON)
+	fi
+
+	if use python; then
+		local targets=( ${PYTHON_TARGETS//[_]/.} )
+		for target in ${targets[@]}; do
+			if [[ "${target}" == *3.* ]]; then
+				mycmakeargs+=(-DWITH_PYTHON3=$(usex python)
+					-DPYTHON3_INCLUDE_DIRS=/usr/include/${target}m
```

Comment:
This seems fragile to me. Did you take a good look at `python-r1` like I asked you earlier? For example, what do you think of https://github.com/hsoft/sandbox-overlay/blob/master/dev-python/python-export/python-export-1.ebuild ? Don't you think that a solution based around `python_export` would be more robust?

Reply:
in fact, my example above is wrong (I've never used this API) and the proper pointer is https://wiki.gentoo.org/wiki/Project:Python/python-utils-r1#Python_build_environment

Reply:
Also, as I see in the API's documentation, `python_is_python3` might work, after all, because it takes an `impl` as an argument. So you could replace your `"${target}" ==` with it and they should be equivalent.

Reply:
I am not sure how to use the `python_is_python3` "efficiently" in this case.

Would 

`if python_is_python3 ${target} ; then`

be considered better(TM) ?

Reply:
Your sarcastic tone doesn't go unnoticed. I understand that you might find this review process heavy, but you're at fault here: I specifically asked you to go look in python-r1 and you silently didn't, submitting it again for review. Yet, it turns out that it would have been a good idea to spend more time looking at python-r1 & friends.

That sloppiness makes reviewers spend more time on your PR than they should. Being sarcastic in a situation where one should be embarrassed might be seen as a lack of judgement.

The `python_is_python3` remark is a "by the way" remark, I wouldn't have made it if it wasn't for the rest. Personally (if it works, I haven't tried it), I would find it more elegant, yes, but I wouldn't reject the ebuild for this reason.

Reply:
I was **not** sarcastic.

Just something to try to lighten the mood.

Reply:
I have implemented everything you asked.

Reply:
Let's clarify your comment earlier.

I **did** look very closely at `python-r1` and found the way to iterate through PYTHON_TARGETS, which was actually super useful, as it allowed me to target the correct implementation. Something that I was not even aware I could do.

However, as I did not see anything related to the library and include files, I just assumed that your suggestion was unrelated to those and did not try to look somewhere else. Nothing more, nothing less.

As a side note, I would recommend avoiding using the _sloppy_ word if you want to stay neutral. This is a **very** aggressive and demeaning word to describe someone else work.

Finally, I do appreciate the overall great work of the Gentoo team, I used this distribution for the past ten year for this reason after all, and yes I do understand the QA process can be painful, hence the attempt to relax the tone.

My clumsy attempts to lighten the mood are now stored in the 'epic' fails box.